Fixing SDL2 video backend implementation; fix Linux builds.

vladikcomper opened the pull request:
This PR fixes initialization of texture for image display (use "SDL_TEXTUREACCESS_STREAMING" flag instead of "SDL_TEXTUREACCESS_STATIC" hint) and incorrect use of "SDL_GetWindowSurface" with the hardware-accelerated renderer.

It effectively fixes emulator crashes on Linux systems (and possibly, other platforms), though some extra testing may be required to retain compatibility with Windows and changing MD video modes by the game itself.

vladikcomper committed:
Fixing various issues and crashes in SDL2 video backend implementation.

This commit fixes initialization of texture for image display (use "SDL_TEXTUREACCESS_STREAMING" flag instead of "SDL_TEXTUREACCESS_STATIC" hint) and incorrect use of "SDL_GetWindowSurface" with the hardware-accelerated renderer.
It effectively fixes emulator crashes on Linux systems (and possibly, other platforms).

vladikcomper committed:
Fix incorrect order of arguments for linker in the Makefile.

Linker flags ($LDFLAGS) should go *after* the objects list, otherwise, specifying shared libraries there may cause linking errors. Also, LDFLAGS themselves are confused with CFLAGS in the command.
For more information, see: https://stackoverflow.com/a/13250090/5551893

vladikcomper committed:
Refactor framerate timing; migrate to "nanosleep", fix time error accumulation due to unreliable implementation.

vladikcomper commented:
This PR was extended with a fix for framerate timing issues.

heyjoeway commented on src/backends/video/video_sdl2.cpp:
```diff
@@ -170,8 +177,7 @@ void Update_Viewport() {
 
-  /* clear destination surface */
-  SDL_FillRect(sdl_winsurf, 0, 0);
+  // TODO: Clear window contents without relying on "SDL_GetWindowSurface".
```
I think SDL_RenderClear will work for that. I'll handle that seperately.

vladikcomper committed:
Fix incorrect casing on "SDL2_image" package name; remove unused "libwebp" dependency.

On Linux, file names are case-sensitive, so the original `pkg-config` command fails, unable to find "sdl2_image" info.

vladikcomper committed:
Remove unused "opusfile" and "libmpg123" dependencies.

When absent on target platform, these dependencies caused pkg-config *and* linking to fail, because the remaining existing dependencies were not added to the options list.

vladikcomper committed:
Split "-L" option from "LIBS" into "LDFLAGS", as recommended by GNU Make manual:

"Non-library linker flags, such as -L, should go in the LDFLAGS variable."

See: https://www.gnu.org/software/make/manual/make.html#Implicit-Variables for more information.

vladikcomper committed:
Fix linking in Makefile again.

LDFLAGS and LIBS were swapped by mistake.

vladikcomper committed:
Remove unused "opusfile" and "libmpg123" dependencies from SDL1 mixer as well.

vladikcomper commented:
Most recent commits also fix issues with using SDL2mixer backend in Linux builds.

vladikcomper committed:
Merge branch 'master' of https://github.com/heyjoeway/Genesis-Plus-GX

heyjoeway commented on src/main.cpp:
```diff
@@ -396,28 +394,37 @@ int main (int argc, char *argv[]) {
-    // uint32 timePrev = 0;
+    timespec timeAfter, timeBefore;
+
     while(running) {
```
Will add a way to use vsync for mainloop timing and fall back to this on refresh rates other than ~60Hz.
